frontend/packages/component-library-styles/README.md

```diff
@@ -1,3 +1,686 @@
+| `--borders-brand-aqua-primary`  | Primary aqua border color   | `var(--brand-aqua-50)` |
+| `--text-brand-aqua-primary`     | Primary aqua text color     | `var(--brand-aqua-50)` |
+
+#### 🌙 Dark Theme Semantic Colors
+
+The dark theme is defined using `[data-theme="Dark"]`.
+
+##### **Example Colors**
+
+| Variable                        | Description                              | Value                         |
+| ------------------------------- | ---------------------------------------- | ----------------------------- |
+| `--background-brand-aqua-light` | Light aqua background color in dark mode | `var(--brand-aqua-90)`        |
+| `--borders-brand-aqua-primary`  | Primary aqua border color in dark mode   | `var(--neutral-gray-50)`      |
+| `--text-brand-aqua-primary`     | Primary aqua text color in dark mode     | `var(--text-neutral-primary)` |
+
+---
+
+#### ✅ Usage Example:
+
+You can use semantic colors directly in your SCSS files:
+
+```scss
+@use '@code-dot-org/component-library-styles/colors.scss';
+
+.myComponent {
+  background-color: var(--background-neutral-primary);
+  color: var(--text-neutral-primary);
+  border: 1px solid var(--borders-neutral-primary);
+}
+```
+
+---
+
+#### ✅ Example with Dark Mode:
+
+To enable dark mode, add a `data-theme="dark"` attribute to the root element:
+
+```jsx
+<div data-theme="dark">
+  <MyComponent />
+</div>
+```
+
+Example SCSS:
+
+```scss
+[data-theme='dark'] {
+  .myComponent {
+    background-color: var(--background-neutral-primary);
+    color: var(--text-neutral-primary);
+  }
+}
+```
+
+---
+
+#### 💡 Best Practices for Semantic Colors:
+
+- ✅ Use **semantic colors** instead of primitive colors whenever possible.
+- ✅ Keep theming consistent by relying on semantic tokens for dark/light mode handling.
+- ✅ For custom styles, ensure semantic tokens are used instead of hard-coded values.
+
+### Primitive Colors
+
+The `primitiveColors.scss` file defines the **base colors** used throughout the design system. Unlike semantic colors,
+primitive colors are **fixed** and do **not change** across light and dark themes.
+
+Primitive colors are meant to be used as the foundation for building semantic colors. They define a consistent color.
+To view the latest primitive color variables, refer to our [DSCO Variables Figma page](https://www.figma.com/design/NIVcvUgU3WmXpAmp9U2vVy/DSCO-Variables?node-id=2925-10156&m=dev).
+palette and ensure a unified visual language across all Code.org components.
+
+#### ✅ Example:
+
+```scss
+@use '@code-dot-org/component-library-styles/primitiveColors.scss';
+
+.myComponent {
+  background-color: var(--brand-aqua-50);
+  color: var(--neutral-gray-90);
+}
+```
+
+---
+
+#### 🌈 Variables Examples
+
+| Variable                  | Description         | Value     |
+| ------------------------- | ------------------- | --------- |
+| `--accent-orange-10`      | Lightest orange     | `#fff6e5` |
```

Comment:
I do slightly worry about hardcoded values like these in documentation falling out of date with actual code (same for font weights below). Is there a good way to prevent that from happening as much as possible? Specific documentation in the style files indicating that any changes must be made in documentation as well? Or some sort of linting? Or even autogenerating some of these tables from code?

Reply:
Yeah, that's a great question.
For variables I'm not sure if it worth the effort due to time restrictions, that is why I added these just as example, without adding all of the possible variants. (assuming one will go to code to see actual variables)

For font related variables/mixins I agree it makes sense  to add a comment to source files. Or maybe even removing them from docs or maybe just show examples instead of whole list?